File: interviews/wizeline/countPalyndromeWordsOnList.py
def countPalyndromes(sampleArray):
    COUNT = {}
    """ 
        var re = /[\W_]/g; // or var re = /[^A-Za-z0-9]/g;
        const CLEAN_WORD = word.toLowerCase().replace(re, '');
    """
    for word in sampleArray:
        REVERSED_WORD = word[::-1]
        if word.lower() == REVERSED_WORD.lower():
            if word not in COUNT:
                COUNT[word] = 1
            else:
                COUNT[word] += 1
    return COUNT
SAMPLE_ARRAY = ['Jorge', 'Ana', 'Ono', 'PPP', 'ASDASD', 'Jorge', 'Ana', 'Ono', 'PPP', 'ASDASD']
RESULT = countPalyndromes(SAMPLE_ARRAY)
print(RESULT)

File: interviews/wizeline/countPalyndromeWordsOnList.js
function countPalyndromes(sampleArray) {
  const COUNT = {};

  sampleArray.forEach(word => {
    /* 
        var re = /[\W_]/g; // or var re = /[^A-Za-z0-9]/g;
        const CLEAN_WORD = word.toLowerCase().replace(re, '');
    */
    const REVERSED_WORD = word.split("").reverse().join("");

    if (word.toLowerCase() === REVERSED_WORD.toLowerCase()) {
      COUNT[word] = COUNT[word] === undefined ? 1 : COUNT[word]++; 
    }
  });
  return COUNT;
};

const SAMPLE_ARRAY = ['Jorge', 'Ana', 'Peace', 'Ono'];
const RESULT = countPalyndromes(SAMPLE_ARRAY);
console.log('[NAVA] RESULT:', RESULT);